jobs/upload-brew.Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+import org.yaml.snakeyaml.Yaml;
+
+def pipeutils, pipecfg, libcloud, uploading, jenkins_agent_image_tag
+def src_config_url, src_config_ref, brew_principle, brew_profile
+node {
+    checkout scm
+    pipeutils = load("utils.groovy")
+    pipecfg = pipeutils.load_pipecfg()
+    libcloud = load("libcloud.groovy")
+
+    brew_principle =  pipecfg['brew']['principle']
+    brew_profile =  pipecfg['brew']['profile']
+}
+
+arches = [ 'aarch64', 'ppc64le', 'x86_64', 's390x']
+
+properties([
+    pipelineTriggers([]),
+    parameters([
+      choice(name: 'ARCH',
+             description: 'The target architecture',
+             choices: (arches),
+             trim: true),
+      string(name: 'VERSION',
+             description: 'Build version',
+             defaultValue: '',
+             trim: true),
+      choice(name: 'STREAM',
+             choices: pipeutils.get_streams_choices(pipecfg),
+             description: 'CoreOS stream to upload to brew'),
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+def build_description = "[${params.VERSION}]"
+
+// Always use the latest
+cosa_img = "quay.io/coreos-assembler/coreos-assembler:latest"
+
+// Note the supermin VM just uses 2G. The really hungry part is xz, which
+// without lots of memory takes lots of time. For now we just hardcode these
+// here; we can look into making them configurable through the template if
+// developers really need to tweak them.
+// XXX bump an extra 2G (to 10.5) because of an error we are seeing in
+// testiso: https://github.com/coreos/fedora-coreos-tracker/issues/1339
+def cosa_memory_request_mb = 10.5 * 1024 as Integer
+
+// Now that we've established the memory constraint based on xz above, derive
+// kola parallelism from that. We leave 512M for overhead and VMs are at most
+// 1.5G each (in general; root reprovisioning tests require 4G which is partly
+// why we run them separately).
+// XXX: https://github.com/coreos/coreos-assembler/issues/3118 will make this
+// cleaner
+def ncpus = ((cosa_memory_request_mb - 512) / 1536) as Integer
```

Comment:
we need to update the comments here since we don't decompress anything OR run `kola`.. I doubt we need as much memory as we do for a full `build` job either.